other/upgrade_2-0_postgresql.sql

```diff
@@ -918,192 +918,67 @@ if (!isset($modSettings['attachment_thumb_png']))
+		// Change back to a string...
+		$known_themes = implode(',', $known_themes);
+
+		// Update the database
 		upgrade_query("
-			DELETE FROM {$db_prefix}settings
-			WHERE variable IN ('theme_default', 'theme_guests', 'knownThemes')");
+			REPLACE INTO {$db_prefix}settings (variable, value)
+			VALUES ('knownThemes', '$known_themes')");
 
-		foreach ($new_settings AS $a_new_setting)
-		{
-			upgrade_query("
-				INSERT INTO {$db_prefix}settings
-				(variable, value)
-				VALUES " . implode(', ', $a_new_setting));
-		}
+		// Delete any info about this theme
+		upgrade_query("
+			DELETE FROM {$db_prefix}themes
+			WHERE id_theme = $id_theme");
 
-		// What about members?
+		// Set any members or boards using this theme to the default
 		upgrade_query("
 			UPDATE {$db_prefix}members
-			SET id_theme = $id_core_theme
-			WHERE id_theme = 1");
+			SET id_theme = 0
+			WHERE id_theme = $id_theme");
 
-		// Boards?
 		upgrade_query("
 			UPDATE {$db_prefix}boards
-			SET id_theme = $id_core_theme
-			WHERE id_theme = 1");
+			SET id_theme = 0
+			WHERE id_theme = $id_theme");
 
-		// The other themes used to use core as their base theme.
-		if (isset($core['theme_dir']) && isset($core['theme_url']))
+		if ($modSettings['theme_guests'] == $id_theme)
 		{
-			$coreBasedThemes = array_diff($allThemes, array(1));
-
-			// Exclude the themes that already have a base_theme_dir.
-			$request = upgrade_query("
-				SELECT DISTINCT id_theme
-				FROM {$db_prefix}themes
-				WHERE variable = 'base_theme_dir'");
-			while ($row = $smcFunc['db_fetch_assoc']($request))
-				$coreBasedThemes = array_diff($coreBasedThemes, array($row['id_theme']));
-			$smcFunc['db_free_result']($request);
-
-			// Only base themes if there are templates that need a fall-back.
-			$insertRows = array();
-			$request = upgrade_query("
-				SELECT id_theme, value AS theme_dir
-				FROM {$db_prefix}themes
-				WHERE id_theme IN (" . implode(', ', $coreBasedThemes) . ")
-					AND id_member = 0
-					AND variable = 'theme_dir'");
-			while ($row = $smcFunc['db_fetch_assoc']($request))
-			{
-				if (!file_exists($row['theme_dir'] . '/BoardIndex.template.php') || !file_exists($row['theme_dir'] . '/Display.template.php') || !file_exists($row['theme_dir'] . '/index.template.php') || !file_exists($row['theme_dir'] . '/MessageIndex.template.php') || !file_exists($row['theme_dir'] . '/Settings.template.php'))
-				{
-					$insertRows[] = "(0, $row[id_theme], 'base_theme_dir', '" . addslashes($core['theme_dir']) . "')";
-					$insertRows[] = "(0, $row[id_theme], 'base_theme_url', '" . addslashes($core['theme_url']) . "')";
-				}
-			}
-			$smcFunc['db_free_result']($request);
-
-			if (!empty($insertRows))
-				upgrade_query("
-					INSERT INTO {$db_prefix}themes
-						(id_member, id_theme, variable, value)
-					VALUES
-						" . implode(',
-						', $insertRows). ' ON CONFLICT DO NOTHING');
+			upgrade_query("
+				REPLACE INTO {$db_prefix}settings
```

Comment:
Use the replace methode of $smcfunc['db_insert'],
replace got a different syntax in pg.

Reply:
Hmmm...  All of the upgrade-helper.php "smf_" functions are mysql specific...  Let me think about this...

Reply:
Replace smcFunc used